src/list/index.tsx

```diff
@@ -153,9 +158,25 @@ export const ListItem = listItemFactory(function ListItem({
 	}
 
 	const [firstChild, ...otherChildren] = children();
-	const { leading = undefined, primary, trailing = undefined } = isRenderResult(firstChild)
-		? { primary: [firstChild, ...otherChildren] }
-		: firstChild;
+	const {
+		leading = undefined,
+		primary,
+		secondary = undefined,
+		trailing = undefined
+	} = isRenderResult(firstChild) ? { primary: [firstChild, ...otherChildren] } : firstChild;
+	const size = height || (secondary ? 'medium' : 'small');
+	const sizeStyle =
```

Comment:
Im not sure we should be setting a size at all, the list items should adapt to the size of their content and render appropriately. In any case we certainly shouldn't be using a massive ternary such as this.